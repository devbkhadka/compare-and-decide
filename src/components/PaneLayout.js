import React from 'react'
import {
    Box,
    makeStyles,
} from '@material-ui/core'

import { joinClss } from '../utils/css_utils'

const useStyles = makeStyles(theme=>({
    root: {
        
    },
    fullHeight: {
        height: '100%'
    },
    leftPanel: {
        width: '250px',
        minWidth: '250px',
        overflow: 'scroll',
        borderRight: `1px solid ${theme.palette.divider}`,
    },
    rightPanel: {
        marginLeft: theme.spacing(2),
        flexGrow: 1
    }
}))

export default ({children})=>{
    const classes = useStyles()
    if(!children || children.length!==2){
        throw Error('PanelLayout should have exactly 2 children, 1st for right pane and 2nd for left pane')
    }
    return <Box display='flex' className={joinClss(classes.root, classes.fullHeight)}>
                <Box display='flex' flexDirection='column' 
                    className={ joinClss(classes.fullHeight, classes.leftPanel)}>
                    {children[0]}
                </Box>
                <Box display='flex' flexDirection='column' 
                    className={ joinClss(classes.fullHeight, classes.rightPanel)}>
                    {children[1]}
                </Box>
            </Box>
}
