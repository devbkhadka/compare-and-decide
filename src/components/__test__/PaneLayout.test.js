import React from 'react'
import {render} from '@testing-library/react'
import PaneLayout from '../PaneLayout'

console.error = jest.fn()
describe('PaneLayout', ()=>{
    it('should raise error if number of children is not exactly two', ()=>{
        expect(()=>render(<PaneLayout></PaneLayout>)).toThrowErrorMatchingSnapshot()
        expect(()=>render(<PaneLayout><div></div></PaneLayout>)).toThrowErrorMatchingSnapshot()
        expect(()=>render(<PaneLayout>
                            <div></div>
                            <div></div>
                            <div></div>
                        </PaneLayout>)).toThrowErrorMatchingSnapshot()
    })

    it('should render children to left and right pane', ()=>{
        const rendered = render(
        <PaneLayout>
            <div>I am left pane</div>
            <div>I am right pane</div>
        </PaneLayout>)

        const leftPane = rendered.getByText('I am left pane')
        const rightPane = rendered.getByText('I am right pane')

        expect(leftPane.parentElement['className']).toContain('leftPanel')
        expect(rightPane.parentElement['className']).toContain('rightPanel')
    })

    test('snapshot is matching', ()=>{
        const rendered = render(
            <PaneLayout>
                <div>I am left pane</div>
                <div>I am right pane</div>
            </PaneLayout>)
        expect(rendered.baseElement).toMatchSnapshot()
    })

    
})