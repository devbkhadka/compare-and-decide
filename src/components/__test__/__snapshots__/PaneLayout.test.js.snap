// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaneLayout should raise error if number of children is not exactly two 1`] = `"PanelLayout should have exactly 2 children, 1st for right pane and 2nd for left pane"`;

exports[`PaneLayout should raise error if number of children is not exactly two 2`] = `"PanelLayout should have exactly 2 children, 1st for right pane and 2nd for left pane"`;

exports[`PaneLayout should raise error if number of children is not exactly two 3`] = `"PanelLayout should have exactly 2 children, 1st for right pane and 2nd for left pane"`;

exports[`PaneLayout snapshot is matching 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root MuiBox-root-10 makeStyles-root-1 makeStyles-fullHeight-2"
    >
      <div
        class="MuiBox-root MuiBox-root-11 makeStyles-fullHeight-2 makeStyles-leftPanel-3"
      >
        <div>
          I am left pane
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-12 makeStyles-fullHeight-2 makeStyles-rightPanel-4"
      >
        <div>
          I am right pane
        </div>
      </div>
    </div>
  </div>
</body>
`;
